pytorch_lightning/trainer/trainer.py

```diff
@@ -1109,6 +1109,9 @@ def _run(
         # ----------------------------
         self._call_callback_hooks("on_before_accelerator_backend_setup")
         self.strategy.setup_environment()
+
+        self.__setup_profiler()
```

Comment:
We could move this even higher, at the beginning of the _run method as ddp_spawn and ddp have been aligned. @awaelchli

Reply:
Yes, I tried it but found the `self._setup_profiler` must called after `self.strategy.setup_environment` and `self.strategy.setup_environment` must called after `self._call_callback_hooks("on_before_accelerator_backend_setup")`, if want move `self._setup_profiler` higher, must move all them three higher. I'm not sure if this will have any problem.